substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2667,7 +2648,7 @@ pub mod pallet {
 		) -> DispatchResult {
 			let mut bonded_pool = match ensure_root(origin.clone()) {
 				Ok(()) => BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?,
-				Err(sp_runtime::traits::BadOrigin) => {
+				Err(_bad_origin) => {
```

Comment:
why not keep the exact match here?
```suggestion
				Err(BadOrigin) => {
```